### PR TITLE
#1403 #1457 #1478 #1487 #1491 #1494  #1501 

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#1457 Shortcuts for commands are not enabled on startup
#1487 REPL is not visible by default in Data Science profile
#1491 PM search results should sort by exact match
#1403 Data import dialog: Encoding dropdown's default is empty
#1501 Switch to Installed second time take forever
#1494 VE: when i double click on a row for a dataframe, it should show the DF
#1478 Package Manager: Don't show "uninstall" icon for system package

### CODE REVIEW

File: src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs [hunk 1]
```diff
@@ -98,13 +99,41 @@ public ImportDataWindow(string filePath, string name)
...
+                }
+                // TODO: map names that R uses to actual encoding names
+                try {
+                    return Encoding.GetEncoding(encoding);
+                } catch (ArgumentException) { }
+            }
+            return Encoding.Default;
+        }
+
+        private void AppendString(StringBuilder builder, string value) {
```
Comment (AlexanderSher) [inline src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs]: Never used

File: src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    Name = "PkgMgr " + g.ToString(),
+                    RBasePath = _settings.RBasePath,
+                    CranMirrorName = _settings.CranMirror
+                }, HostStartTimeout);
+            }
+            return session;
+        }
+    }
+
+    internal sealed class SessionToken : IDisposable {
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs]: Never used

File: src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                    }
+                    s.Dispose();
+                }
+                return session ?? await CreatePackageQuerySessionAsync();
+            } finally {
+                _sem.Release();
+            }
+        }
+
+        public async Task ReleaseSession(IRSession session) {
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs]: Never used

File: src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+        public async Task<IRSession> GetSession() {
+            if (_disposed) {
+                throw new InvalidOperationException("Session pool is disposed");
+            }
+
+            await _sem.WaitAsync();
+            try {
+                IRSession session = null;
+                while (_freeSessions.Count > 0) {
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs]: There is no code that is called from anywhere and adds items to _freeSessions.

Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs]: SessionToken does

File: src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs]: Merge artifact, deleted

File: src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs (hunk 4 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs]: Oops, need to return that one actually

File: src/R/Components/Impl/PackageManager/ViewModel/IRPackageViewModel.cs [hunk 5]
```diff
@@ -28,6 +28,7 @@ public interface IRPackageViewModel {
         bool IsInstalled { get; set; }
         bool IsLoaded { get; set; }
         bool IsUpdateAvailable { get; }
+        bool CanUninstall { get; set; }
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/ViewModel/IRPackageViewModel.cs]: So for now we don't use this property in XAML, correct?

File: src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs (hunk 3 above)
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/SessionPool.cs]: Can be private

File: src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs [hunk 6]
```diff
@@ -165,46 +150,42 @@ internal class RPackageManager : IRPackageManager {
...
-
-            try {
-                // Get the repos and libpaths from the REPL session and set them
-                // in the package query session
-                await EvalRepositoriesAsync(await DeparseRepositoriesAsync());
-                await EvalLibrariesAsync(await DeparseLibrariesAsync());
-
-                var result = await queryFunc(_pkgQuerySession);
-                CheckEvaluationResult(result);
+            using (var sessionToken = await _sessionPool.GetSession()) {
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs]: `try/catch` should wrap `using (var sessionToken = await _sessionPool.GetSession())`, not vice versa

**Comment by AlexanderSher** [comment]
:+1:

**Pull request merged**